src/main.cpp

```diff
@@ -4529,31 +4516,24 @@ bool SendMessages(CNode* pto, bool fSendTrickle)
         if (!lockMain)
             return true;
 
-        // Address refresh broadcast
-        static int64_t nLastRebroadcast;
-        if (!IsInitialBlockDownload() && (GetTime() - nLastRebroadcast > 24 * 60 * 60))
-        {
-            LOCK(cs_vNodes);
-            BOOST_FOREACH(CNode* pnode, vNodes)
-            {
-                // Periodically clear setAddrKnown to allow refresh broadcasts
-                if (nLastRebroadcast)
-                    pnode->setAddrKnown.clear();
-
-                // Rebroadcast our address
-                AdvertizeLocal(pnode);
-            }
-            if (!vNodes.empty())
-                nLastRebroadcast = GetTime();
-        }
-
         //
         // Message: addr
         //
-        if (fSendTrickle)
-        {
+
+        if (pto->nNextLocalAddrSend < GetTime()) {
+            AdvertizeLocal(pto);
+            pto->nNextLocalAddrSend = GetTime() + GetRand(24 * 60 * 60);
```

Comment:
Could that magic number be defined somehwere as constant? IMHO it's used quite often in this pull.